src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/features/DocumentProcessorTest.java

```diff
@@ -0,0 +1,90 @@
+package uk.gov.hmcts.reform.bulkscanprocessor.features;
+
+import com.microsoft.azure.storage.CloudStorageAccount;
+import com.microsoft.azure.storage.StorageException;
+import com.microsoft.azure.storage.blob.CloudBlobClient;
+import com.microsoft.azure.storage.blob.CloudBlobContainer;
+import com.microsoft.azure.storage.blob.CloudBlockBlob;
+import com.palantir.docker.compose.DockerComposeRule;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+import uk.gov.hmcts.reform.bulkscanprocessor.services.DocumentProcessor;
+import uk.gov.hmcts.reform.bulkscanprocessor.services.PDF;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.security.InvalidKeyException;
+import java.util.List;
+import java.util.function.Consumer;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.catchThrowable;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+@RunWith(MockitoJUnitRunner.class)
+public class DocumentProcessorTest {
+
+    @ClassRule
+    public static DockerComposeRule docker = DockerComposeRule.builder()
+        .file("src/integrationTest/resources/docker-compose.yml")
+        .build();
+
+    @Mock
+    private Consumer<List<PDF>> pdfsConsumer;
+
+    @Captor
+    private ArgumentCaptor<List<PDF>> pdfListCaptor;
+
+    private CloudBlobClient cloudBlobClient;
+
+    @Before
+    public void setup() throws URISyntaxException, InvalidKeyException, StorageException {
+        CloudStorageAccount account = CloudStorageAccount.parse("UseDevelopmentStorage=true");
+        cloudBlobClient = account.createCloudBlobClient();
+    }
+
+    @Test
+    public void connects_without_exceptions() throws URISyntaxException, StorageException {
+        CloudBlobContainer abc = cloudBlobClient.getContainerReference("abc");
+        abc.createIfNotExists();
+
+        Throwable throwable = catchThrowable(() -> {
+            DocumentProcessor documentProcessor = new DocumentProcessor(cloudBlobClient, pdfsConsumer);
+            documentProcessor.readBlobs();
+        });
+
+        assertThat(throwable).doesNotThrowAnyException();
```

Comment:
You have to put the lambda as the argument, not the result. And shouldn't it be assertThatCode(...?

Or expect the throwable to be null.